Content.Client/_DV/UserInterfaces/Systems/Cwoink/CHelpUIController.cs

```diff
@@ -0,0 +1,558 @@
+{
+    [Dependency] private readonly IClientAdminManager _adminManager = default!;
+    [Dependency] private readonly IConfigurationManager _config = default!;
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IClyde _clyde = default!;
+    [Dependency] private readonly IUserInterfaceManager _uiManager = default!;
+    [UISystemDependency] private readonly AudioSystem _audio = default!;
+
+    private CwoinkSystem? _cwoinkSystem;
+    private MenuButton? GameCHelpButton => UIManager.GetActiveUIWidgetOrNull<GameTopMenuBar>()?.CHelpButton;
+    public ICHelpUIHandler? UIHelper;
+    private bool _discordRelayActive;
+    private bool _hasUnreadCHelp;
+    private bool _cwoinkSoundEnabled;
+    private string? _cHelpSound;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeNetworkEvent<CwoinkDiscordRelayUpdated>(DiscordRelayUpdated);
+        SubscribeNetworkEvent<CwoinkPlayerTypingUpdated>(PeopleTypingUpdated);
+
+        _adminManager.AdminStatusUpdated += OnAdminStatusUpdated;
+        _config.OnValueChanged(DCCVars.CHelpSound, v => _cHelpSound = v, true);
+        _config.OnValueChanged(CCVars.BwoinkSoundEnabled, v => _cwoinkSoundEnabled = v, true);
+    }
+
+    public void UnloadButton()
+    {
+        if (GameCHelpButton != null)
+            GameCHelpButton.OnPressed -= CHelpButtonPressed;
+    }
+
+    public void LoadButton()
+    {
+        if (GameCHelpButton != null)
+            GameCHelpButton.OnPressed += CHelpButtonPressed;
+    }
+
+    private void OnAdminStatusUpdated()
+    {
+        if (UIHelper is not { IsOpen: true })
+            return;
+        EnsureUIHelper();
+    }
+
+    private void CHelpButtonPressed(BaseButton.ButtonEventArgs obj)
+    {
+        EnsureUIHelper();
+        UIHelper!.ToggleWindow();
+    }
+
+    public void OnSystemLoaded(CwoinkSystem system)
+    {
+        _cwoinkSystem = system;
+        _cwoinkSystem.OnCwoinkTextMessageRecieved += ReceivedCwoink;
+
+        CommandBinds.Builder
+            .Bind(ContentKeyFunctions.OpenCHelp,
+                InputCmdHandler.FromDelegate(_ => ToggleWindow()))
+            .Register<CHelpUIController>();
+    }
+
+    public void OnSystemUnloaded(CwoinkSystem system)
+    {
+        CommandBinds.Unregister<CHelpUIController>();
+
+        DebugTools.Assert(_cwoinkSystem != null);
+        _cwoinkSystem!.OnCwoinkTextMessageRecieved -= ReceivedCwoink;
+        _cwoinkSystem = null;
+    }
+
+    private void SetCHelpPressed(bool pressed)
+    {
+        if (GameCHelpButton != null)
+        {
+            GameCHelpButton.Pressed = pressed;
+        }
+
+        UIManager.ClickSound();
+        UnreadCHelpRead();
+    }
+
+    private void ReceivedCwoink(object? sender, CwoinkTextMessage message)
+    {
+        Logger.InfoS("c.s.go.es.cwoink", $"@{message.UserId}: {message.Text}");
```

Comment:
Don't use obsolete method
Also is this sawmill name normal?
```suggestion
        Logger.GetSawmill("c.s.go.es.cwoink").Info($"@{message.UserId}: {message.Text}");
```